registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java

```diff
@@ -368,7 +369,19 @@ private ResponseDTO save(List<BiometricsDto> biometrics) {
 		LOGGER.info(LOG_REG_USER_ONBOARD, APPLICATION_NAME, APPLICATION_ID, "Entering save method");
 
 		try {
-			onBoardingResponse = userOnBoardDao.insert(biometrics);
+			List<BIR> birList = new ArrayList<>();
+			for (BiometricsDto biometricsDto : biometrics) {
+				BIR bir = buildBir(biometricsDto);
+				LOGGER.debug(LOG_REG_USER_ONBOARD, APPLICATION_NAME, APPLICATION_ID, "Adding bir");
+				birList.add(bir);
+			}
+
+			List<BIR> templates = bioAPIFactory
+					.getBioProvider(BiometricType.fromValue(birList.get(0).getBdbInfo().getType().get(0).value()),
```

Comment:
You are taking BiometricType of first element and passing biometrics of other biometricType as well to extract method. We should get bioPriovder for each BiometricType separately and then invoke extract with biometrics related to corresponding BiometricType.

Reply:
Modified to get bioProvider for each BiometricType separately